kDrive/AppRestorationService.swift

```diff
@@ -45,44 +47,31 @@ public final class AppRestorationService: AppRestorationServiceable {
     @LazyInjectService private var accountManager: AccountManageable
 
     /// State restoration version
-    private static let currentStateVersion = 4
-
-    /// State restoration key
-    private static let appStateVersionKey = "appStateVersionKey"
+    private static let currentStateVersion = 5
 
     public init() {
         // META: keep SonarCloud happy
     }
 
-    public func shouldSaveApplicationState(coder: NSCoder) -> Bool {
+    public var shouldSaveApplicationState: Bool {
         Log.sceneDelegate("shouldSaveApplicationState")
         Log.sceneDelegate("Restoration files:\(String(describing: Self.statePath))")
-        coder.encode(Self.currentStateVersion, forKey: Self.appStateVersionKey)
-        return true
-    }
 
-    public func shouldRestoreApplicationState(coder: NSCoder) -> Bool {
-        Log.sceneDelegate("shouldRestoreApplicationState coder:")
         return true
-        /* let encodedVersion = coder.decodeInteger(forKey: Self.appStateVersionKey)
-         let shouldRestoreApplicationState = Self.currentStateVersion == encodedVersion &&
-             !(UserDefaults.shared.legacyIsFirstLaunch || accountManager.accounts.isEmpty)
-         Log.appDelegate("shouldRestoreApplicationState:\(shouldRestoreApplicationState)")
-         return shouldRestoreApplicationState */
     }
 
     public var shouldRestoreApplicationState: Bool {
-        Log.sceneDelegate("shouldRestoreApplicationState")
-        return true
+        let storedVersion = UserDefaults.shared.appRestorationVersion
+        let shouldRestore = Self.currentStateVersion == storedVersion &&
+            !(UserDefaults.shared.legacyIsFirstLaunch || accountManager.accounts.isEmpty)
+
+        Log.sceneDelegate("shouldRestoreApplicationState:\(shouldRestore) appRestorationVersion:\(storedVersion)")
+        return shouldRestore
+    }
 
-        // TODO: Use user defaults
-        /*
-         let encodedVersion = coder.decodeInteger(forKey: Self.appStateVersionKey)
-         let shouldRestoreApplicationState = Self.currentStateVersion == encodedVersion &&
-             !(UserDefaults.shared.legacyIsFirstLaunch || accountManager.accounts.isEmpty)
-         Log.appDelegate("shouldRestoreApplicationState:\(shouldRestoreApplicationState)")
-         return shouldRestoreApplicationState
-          */
+    public func saveRestorationVersion() {
+        UserDefaults.shared.appRestorationVersion = Self.currentStateVersion
```

Comment:
`currentStateVersion` is stored in `core` under another key in user defaults.
The first startup of this version will not enable State restoration